File: codex_poting/java/src/main/java/com/example/demo/api/request/BookSalesUnitPriceCreateRequest.java
package com.example.demo.api.request;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Value;

import java.time.LocalDate;

@Value
public class BookSalesUnitPriceCreateRequest {
    @NotNull
    @Min(1)
    @Max(10000)
    @Schema(description = "販売単価", type = "integer", example = "1500")
    Integer salesUnitPrice;

    @NotNull
    @Future
    @Schema(description = "有効開始日", type = "string", format = "date")
    LocalDate effectiveFrom;
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/api/request/BookSalesUnitPriceCreateRequest.kt
package com.example.demo.api.request

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.Future
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull
import java.time.LocalDate

data class BookSalesUnitPriceCreateRequest(
    @field:NotNull
    @field:Min(1)
    @field:Max(10000)
    @field:Schema(description = "販売単価", type = "integer", example = "1500")
    val salesUnitPrice: Int?,

    @field:NotNull
    @field:Future
    @field:Schema(description = "有効開始日", type = "string", format = "date")
    val effectiveFrom: LocalDate?
)
